pkg/controller/sparkapplication/controller.go

```diff
@@ -515,12 +508,18 @@ func (c *Controller) syncSparkApplication(key string) error {
 				if completionTime != nil {
 					appToUpdate.Status.SubmissionTime = *completionTime
 				}
+				c.recordSparkApplicationEvent(appToUpdate)
 			} else {
 				// Since we delegate submission retries to the Kubernetes Job controller, the fact that the
 				// submission Job failed means all the submission attempts failed. So we set the application
 				// state to FailedSubmission, which is a terminal state.
 				appToUpdate.Status.AppState.State = v1beta1.FailedSubmissionState
+				appToUpdate.Status.AppState.ErrorMessage = err.Error()
```

Comment:
`err` may be `nil` and cause an panic here.

Reply:
shouldn't be but happy to add a nil check